### PR TITLE
Events page added

### PR DESCRIPTION
**Pull request by bmegha98**

### CODE REVIEW

**Comment by rumansaleem** [comment]
we need to list all the events on the event page as well. Moreover, you should use a sperate layout that extends base layout for events, which will make use of the yaml variables, such as image, title etc.

**Comment by bmegha98** [comment]
Ya i have done that in event.ejs file by making use of image variable

**Comment by rumansaleem** [comment]
But it is not reflected on the event's page (for example: on `/events/alacrity`), and what about the listing of events on `/events` page.

File: pages/events/codingclub.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+layout : base-layout.ejs
+title : CodingClub
+tags: ["event"]
+permalink : /events/Coding Club/
```
Comment (rumansaleem) [inline pages/events/codingclub.md]: it should be sluggish cased (all lowercase, words separated by a hyphen), `/events/coding-club`

File: pages/events/chakravyuh.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+layout : base-layout.ejs
+title : Chakravyuh
+tags: ["event"]
+permalink : /events/Chakravyuh/
```
Comment (rumansaleem) [inline pages/events/chakravyuh.md]: make sure all the permalinks are lower cased and words separated by hyphens.

File: pages/events/brainspark.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+layout : base-layout.ejs
+title : Brainspark
+tags: ["event"]
```
Comment (rumansaleem) [inline pages/events/brainspark.md:4]: you can add tags like, "puzzle", "mind game" for now it doesn't matter. But you got the idea.

File: _includes/events.ejs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                <h1 class = "banner-subtitle">Events</h1>
+                <h2 class = "banner-title">Something for everyone !</h2>
+                <p>
+                    "We have 9 technical events as well as 7 non-technical events."
+                </p>
+            </div>
+        </div>
+        <div class = "row">
+            <div class = "grid-container">
+                <% events.forEach(function(event){ %>
```
Comment (rumansaleem) [inline _includes/events.ejs:29]: seems like it doesn't work, or the zeit didnt built this correctly.

File: _includes/events.ejs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <link href="/assets/stylesheets/events.css" rel="stylesheet"/>
+    </head>
+    <body>
+        <div class = "banner">
+            <div class = "banner-content">
+                <h1 class = "banner-subtitle">Events</h1>
+                <h2 class = "banner-title">Something for everyone !</h2>
+                <p>
+                    "We have 9 technical events as well as 7 non-technical events."
```
Comment (rumansaleem) [inline _includes/events.ejs:23]: make these counts dynamic, use a front-matter boolean variable `technical` for all events, filter the array of events and display their counts dynamically. You can also segregate the technical and non-technical events in the listings.

File: _includes/events.ejs [hunk 6]
```diff
@@ -0,0 +1,40 @@
+---
+layout: base-layout.ejs
+pagination:
```
Comment (rumansaleem) [inline _includes/events.ejs:3]: I don't think pagination is required here... We would list all the events on the same page they are no more than 25 (though they are only 8-10). Its completely fine to list them all.

File: pages/events/alfaaz.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+---
+layout : base-layout.ejs
+title : Alfaaz
+tags: ["event"]
+permalink : /events/Alfaaz_Aur_Jazbaat/
```
Comment (rumansaleem) [inline pages/events/alfaaz.md]: make it `/alfaaz-aur-jazbaat`

File: pages/events/codingclub.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+---
+layout : base-layout.ejs
+title : CodingClub
```
Comment (rumansaleem) [inline pages/events/codingclub.md]: I think its coding "cup", and you should add a space here.

**Review by rumansaleem — CHANGES_REQUESTED** [review]
Please make the suggested changes and make sure your build lists all the events on the listings page and they are properly linked to the each event's page. Also, make sure event page shows posters.

**Comment by bmegha98** [comment]
Krishna told us that some of the events might get combined so for now i have listed only few events and can later change the events accordingly and how do i list all events on particular event page. All the events will be listed on main event page no?

**Comment by rumansaleem** [comment]
@bmegha98 It's okay you can change events later. But right now I cannot see any events listed on `/events` page. Here is a screenshot of the `/events` page on build environment:

sankalan-2020-events-page

**Comment by bmegha98** [comment]
How's it possible :|
It's working on my system 
Will look into this

**Comment by rumansaleem** [comment]
if its working on your system, then maybe there's some problem with the Zeit's build. Let me test this locally on my system.

**UPDATE**
I looked into the issue, you have `pages/events.md` in the root which has only static content, that is being build into `events.html` page. Also, the page where you list all the events (`_includes/events.ejs`) is inside `_includes` directory, `_includes` directory is for partials and templates, not for pages, these pages are not built independently.

You should remove `pages/events.md` and move `_includes/events.ejs` to root directory besides, `index.ejs`.

**Comment by bmegha98** [comment]
Ohkay
Will make other changes

**Comment by rumansaleem** [comment]
Please refer to the previous reply.

**Comment by rumansaleem** [comment]
there should be a poster displayed on the event page as well, right now its empty.
Screenshot from 2019-12-28 19-09-04

**Review by rumansaleem — APPROVED** [review]

**Comment by bmegha98** [comment]
yess ! i haven't made layout for event pages yet

**Comment by rumansaleem** [comment]
okay let's keep it here, I'm merging these changes. But start working on responsiveness.

**Pull request merged**